examples/solvers/adupdates_tomography.py

```diff
@@ -0,0 +1,149 @@
+:math:`m` parts and solve the least squares problem
+
+..math::
+    \min_{x\in\mathbb{R}^n} D(x) := \sum_{j = 1}^m \| A_j x - y_j \|^2.
+
+To regularize, we add terms for the total variation and for a functional which
+guarantees that the solution will be pointwise nonnegative. The variational
+regularization problem we are solving is therefore
+
+..math::
+    \min_{x\in \mathbb{R}^n_+} D(x) + \sum_{i = 1}^d (\| \partial_{i, 1} x \|_1
+    + \| \partial_{i, 2} x \|_1.
+
+Here, :math:`\partial_{i, 1}` and :math:`\partial_{i, 2}` contain the even and
+odd components, respectively, of the discretized :math:`i`th partial
+derivative, and :math:`d` is the dimension of the tomography problem. In this
+example, :math:`d = 2`. We solve the problem with the alternating dual updates
+method. For further details, see
+`[MF2015] <http://ieeexplore.ieee.org/document/7271047/>`_.
+
+References
+----------
+[MF2015] McGaffin, M G, and Fessler, J A. *Alternating dual updates
+algorithm for X-ray CT reconstruction on the GPU*. IEEE Transactions
+on Computational Imaging, 1.3 (2015), pp 186--199.
+"""
+
+import numpy as np
+import odl
+
+# The following parameters determine how to split the data (sinograms) and if
+# the solver should do the inner iterations in a fixed order or at random.
+SPLIT_METHOD = 'interlaced'  # How to split the data ('block' or 'interlaced')?
+SPLIT_NUMBER = 20     # How many pieces of data?
+RANDOM = True         # Choose the oder of the inner iterations at random?
+
+# --- Create simulated data (phantom) ---
+
+# Reconstruction space: Set of two-dimensional quadratic images with
+# given resolution.
+reconstruction_space = odl.uniform_discr(min_pt=[-40.0, -40.0],
+                                         max_pt=[40.0, 40.0],
+                                         shape=[1024, 1024])
+phantom = odl.phantom.shepp_logan(reconstruction_space, modified=True)
+
+# Create the forward operators. They correspond to a fully sampled parallel
+# beam geometry.
+geometry = odl.tomo.parallel_beam_geometry(reconstruction_space)
+
+if SPLIT_METHOD == 'block':
+    # Split the data into blocks:
+    # 111 222 333
+    ns = geometry.angles.size // SPLIT_NUMBER
+    ray_trafos = [odl.tomo.RayTransform(reconstruction_space,
+                                        geometry[i * ns:(i + 1) * ns])
+                  for i in range(SPLIT_NUMBER)]
+
+elif SPLIT_METHOD == 'interlaced':
+    # Split the data into slices:
+    # 123 123 123
+    ray_trafos = [odl.tomo.RayTransform(reconstruction_space,
+                                        geometry[i::SPLIT_NUMBER])
+                  for i in range(SPLIT_NUMBER)]
+else:
+    raise ValueError('unknown data split`{}`: typo?'.format(SPLIT_METHOD))
+
+# Create the artificial data.
+data_spaces = [op.range for op in ray_trafos]
+data = [op(phantom) + 5.0 * odl.phantom.white_noise(ds)
+        for (op, ds) in zip(ray_trafos, data_spaces)]
```

Comment:
It's fine like this, but if you want, you can use a fixed *relative* amount of noise (for some notion of "amount"), e.g.
```python
noisefree_data = [op(phantom) for op in ray_trafos]
data = [proj + 0.05 * np.ptp(proj) * odl.phantom.white_noise(proj.space) for proj in noisefree_data]
```
(`np.ptp` is the one-function version of `max - min` that I didn't know until recently)